homeassistant/util/__init__.py

```diff
@@ -311,16 +318,22 @@ def wrapper(*args, **kwargs):
             if not throttle[0].acquire(False):
                 return throttled_value()
 
-            # Check if method is never called or no_throttle is given
-            force = kwargs.pop('no_throttle', False) or not throttle[1]
-
             try:
+                if self.immediate_throttle:
+                    # This only applies the very first time
+                    throttle[1] = utcnow()
+                    self.immediate_throttle = False
+                    return throttled_value()
+
+                # Check if method is never called or no_throttle is given
+                force = kwargs.pop('no_throttle', False) or not throttle[1]
+
                 if force or utcnow() - throttle[1] > self.min_time:
-                    result = method(*args, **kwargs)
```

Comment:
This was done on purpose. Please revert

Reply:
OK. What is the use case for not respecting the regular interval?

Reply:
From the documentation of this class:

>     This method decorator adds a cooldown to a method to prevent it from being
>     called more then 1 time within the timedelta interval `min_time` after it
>     returned its result.

If an update hangs and takes a minute, you don't want to just pile up more calls. You want to wait till the update is done.

Reply:
OK, I see. That could be avoided but anyway I'm fine reverting this, it's not the point of this PR.